org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/AEnv.java

```diff
@@ -429,6 +438,51 @@ public static void actionZoom(Lookup lookup, Object value)
         }
     }
 
+    /**
+	 *  Opens the Drill Assistant
+	 * 	@param data query
+	 *  @param component
+	 */
+    public static void actionDrill(DrillData data, int windowNo) {
+    	int AD_Table_ID = MTable.getTable_ID(data.getQuery().getTableName());
+		if (!MRole.getDefault().isCanReport(AD_Table_ID))
+		{
+			Dialog.error(windowNo, "AccessCannotReport", data.getQuery().getTableName());
+			return;
+		}
+		if (AD_Table_ID != 0) {
+			WDrillReport drillReport = new WDrillReport(data, windowNo);
+
+			Object window = SessionManager.getAppDesktop().findWindow(windowNo);
+			if (window != null && window instanceof Component && window instanceof ISupportMask){
+				final ISupportMask parent = LayoutUtils.showWindowWithMask(drillReport, (Component)window, LayoutUtils.OVERLAP_PARENT);
+				drillReport.addEventListener(DialogEvents.ON_WINDOW_CLOSE, new EventListener<Event>() {
+					@Override
+					public void onEvent(Event event) throws Exception {
+						parent.hideMask();
+					}
+				});
+			}else if (window != null && window instanceof Component){
+				final Mask mask = LayoutUtils.showWindowWithMask(drillReport, (Component)window, null);
+				drillReport.addEventListener(DialogEvents.ON_WINDOW_CLOSE, new EventListener<Event>() {
+					@Override
+					public void onEvent(Event event) throws Exception {
+						mask.hideMask();
+					}
+				});
+			}else{
+				// Add proper width
+				int width = SessionManager.getAppDesktop().getClientInfo().screenWidth * 42 / 100;
+				drillReport.setWidth(width + "px");
+				drillReport.setPosition("center");
+				drillReport.setAttribute(Window.MODE_KEY, Window.MODE_POPUP);
+				AEnv.showWindow(drillReport);
+			}
```

Comment:
perhaps this block can be simplify to just drillReport.doHighlighted().

Reply:
I've tried drillReport.doHighlighted() and drillReport.doPopup() but it didn't open the assistant

Reply:
I've tried drillReport.doHighlighted() and drillReport.doPopup(), but it didn't open the drill assistant.

Reply:
you can set mode to Window.MODE_HIGHLIGHTED and call AEnv.showWindow(drillReport).

Reply:
Yes, that works, thank you

Reply:
actually my doubt is - beside the aenv.showWindow() block, I think we don't need the 2 other block that uses layoututil. also, it seems inconsistent that you set explicit size in the showwindow block but didn't do that in the layoututil block, is the setting of explicit size really needed here ? and if it is really needed, you need to cater for mobile, 42% of screen width wouldn't works well on phone.

Reply:
I tested, setting the screen size is probably unnecessary, it works fine on phone too without it - removed it in the latest commit

The two other if else blocks should be there because this method is called from ZkReportViewer too (I've removed the code from there in the first commit, so we don't have duplicate), and when its called from there, it uses the first layoututil block